Document HTTP JSON API authentication (#3657)
digital-asset/daml: docs/source/json-api/index.rst

            prefix -- URL prefix,
            directory -- local directory that will be mapped to the URL prefix.
            Example: "prefix=static,directory=./static-content"
      --access-token-file <value>
        provide the path from which the access token will be read, required to interact with an authenticated ledger, no default

With Authentication
===================

> If you are running the HTTP JSON API against a Ledger API server that requires authentication, you must provide the access token when you start it.

Reviewer: The last `it` may refer to either the _HTTP JSON API_ or the _Ledger API Server_

Author: Addressed by https://github.com/digital-asset/daml/pull/3657/commits/1bfdf8281ba0f599cc4a87d595f2afd120f8c413

> The access token retrieval depends on the specific DAML setup you are working with: please refer to the ledger operator to learn how.

Reviewer: ```suggestion
The access token retrieval depends on the specific DAML setup you are working with: please refer to the ledger operator to learn how to retrieve a token.
```

Once you have retrieved your access token, you can provide it to the HTTP JSON API by storing it in a file and provide the path to it using the ``--access-token-file`` command line option.

> Both in the case in which the token cannot be read from the provided path or if the Ledger API reports an authentication error (for example due to token expiration), the HTTP JSON API will report the error via logging. The token file can be updated with a valid token and it will be picked up at the next attempt.

Reviewer: ```suggestion
If the token cannot be read from the provided path or the Ledger API reports an authentication error (for example due to token expiration), the HTTP JSON API will report the error via logging. The token file can be updated with a valid token and it will be picked up at the next attempt to send a request.
```

> Note that this token is used exclusively for the HTTP JSON API server and that it will not be use to authenticate client calls to the HTTP JSON API: the user is expected to provide a valid authentication token with each call.

Reviewer: ```suggestion
Note that this token is used exclusively for the HTTP JSON API server and that it will not be used to authenticate client calls to the HTTP JSON API: the user is expected to provide a valid authentication token with each call.
```

Reviewer: It is not clear up to this point why the HTTP JSON API needs to be started with a token on its own that doesn't necessarily represent claims to a party (and probably it should be advised to not provide such a token?).

Author: Addressed by https://github.com/digital-asset/daml/pull/3657/commits/1bfdf8281ba0f599cc4a87d595f2afd120f8c413

            prefix -- URL prefix,
            directory -- local directory that will be mapped to the URL prefix.
            Example: "prefix=static,directory=./static-content"
      --access-token-file <value>
        provide the path from which the access token will be read, required to interact with an authenticated ledger, no default

With Authentication
===================

Apart from interacting with the Ledger API on behalf of the user, the HTTP JSON API server must also interact with the Ledger API to maintain some relevant internal state.

For this reason, you must provide an access token when you start the HTTP JSON API if you're running it against a Ledger API server that requires authentication.

> Note that this token is used exclusively for the HTTP JSON API server and that it will not be use to authenticate client calls to the HTTP JSON API: the user is expected to provide a valid authentication token with each call.

Reviewer 2: ```suggestion
Note that this token is used exclusively for maintaining the internal list of known packages and templates, and that it will not be use to authenticate client calls to the HTTP JSON API: the user is expected to provide a valid authentication token with each call.
```

> The HTTP JSON API servers requires no access to party-specific data and requires exclusively access to the ledger identity and package services: a token issued for the HTTP JSON API server should contain enough claims to contact these two service but no more than that. Please refer to your ledger operator's documentation to find out how.

Reviewer 2: ```suggestion
The HTTP JSON API servers requires no access to party-specific data, only access to the ledger identity and package services: a token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that. Please refer to your ledger operator's documentation to find out how.
```
